src/common/Smi.Common/RabbitMQAdapter.cs

```diff
@@ -135,19 +137,41 @@ public Guid StartConsumer(ConsumerOptions consumerOptions, IConsumer consumer, b
                 throw new ApplicationException("Already a consumer on queue " + consumerOptions.QueueName + " and solo consumer was specified");
             }
 
-            Subscription subscription;
+            Subscription subscription = null;
+            var connected = false;
+            var failed = 0;
 
-            try
-            {
-                subscription = new Subscription(model, consumerOptions.QueueName, consumerOptions.AutoAck, label);
-            }
-            catch (Exception e)
+            while (!connected)
             {
-                model.Close(200, "StartConsumer - Couldn't create subscription");
-                connection.Close(200, "StartConsumer - Couldn't create subscription");
+                try
+                {
+                    subscription = new Subscription(model, consumerOptions.QueueName, consumerOptions.AutoAck, label);
+                    connected = true;
+                }
+                catch (TimeoutException)
+                {
+                    if (++failed >= MaxSubscriptionAttempts)
+                    {
+                        _logger.Warn("Retries exceeded, throwing exception");
+                        throw;
+                    }
 
-                throw new ApplicationException(
-                    "Error when creating subscription on queue \"" + consumerOptions.QueueName + "\"", e);
+                    _logger.Warn($"Timeout when creating Subscription, retrying in 5s...");
+                    Thread.Sleep(TimeSpan.FromSeconds(5));
```

Comment:
Should we make this a random amount of time so that we don't get them all retrying at the same time?

Reply:
We could yes, let's see how this goes first.